File: CSharp/Tests/RepeatingLettersTest.cs
using Xunit;

namespace CSharp.Tests
{
    public class RepeatingLettersTest
    {
        [Theory]
        [InlineData("String", "SSttrriinngg")]
        [InlineData("Hello World!", "HHeelllloo  WWoorrlldd!!")]
        [InlineData("1234!_ ", "11223344!!__  ")]
        [InlineData("##^&%%*&%%$#@@!", "####^^&&%%%%**&&%%%%$$##@@@@!!")]
        [InlineData("scandal", "ssccaannddaall")]
        [InlineData("economics", "eeccoonnoommiiccss")]
        [InlineData(" ", "  ")]
        [InlineData("_______", "______________")]
        [InlineData("equip gallon read", "eeqquuiipp  ggaalllloonn  rreeaadd")]
        [InlineData("baby increase", "bbaabbyy  iinnccrreeaassee")]
        public void DoubleChar_StringInputValue_ReturnStringWithEachCharacterRepeated(string str, string expected)
        {
            var actual = RepeatingLetters.DoubleChar(str);

            Assert.Equal(expected, actual);
        }
    }
}

File: js/tests/repeatingLetters.test.js
const doubleChar = require("../repeatingLetters");

const data = [
  ["String", "SSttrriinngg"],
  ["Hello World!", "HHeelllloo  WWoorrlldd!!"],
  ["1234!_ ", "11223344!!__  "],
  ["##^&%%*&%%$#@@!", "####^^&&%%%%**&&%%%%$$##@@@@!!"],
  ["scandal", "ssccaannddaall"],
  ["economics", "eeccoonnoommiiccss"],
  [" ", "  "],
  ["_______", "______________"],
  ["equip gallon read", "eeqquuiipp  ggaalllloonn  rreeaadd"],
  ["baby increase", "bbaabbyy  iinnccrreeaassee"]
];

describe("doubleChar", () => {
  describe("Return string with characters reapeated once", () => {
    test.each(data)("doubleChar(%p) = %p", (str, result) => {
      var actual = doubleChar(str);

      expect(actual).toEqual(result);
    });
  });
});
